src/Introspection.hack

```diff
@@ -0,0 +1,289 @@
+final class __Field {
+    public function __construct(private IntrospectableField $inner_type) {}
+
+    <<\Slack\GraphQL\Field('name', 'Name of the field')>>
+    public function getName(): string {
+        return $this->inner_type->getName();
+    }
+
+    <<\Slack\GraphQL\Field('description', 'Description of the field')>>
+    public function getDescription(): ?string {
+        return $this->inner_type->getDescription();
+    }
+
+    <<\Slack\GraphQL\Field('args', 'Args of the field')>>
+    public function getArgs(): vec<__InputValue> {
+        return Vec\map($this->inner_type->getArgs(), $arg ==> new __InputValue($arg));
+    }
+
+    <<\Slack\GraphQL\Field('type', 'Type of the field')>>
+    public function getType(): __Type {
+        return new __Type($this->inner_type->getType());
+    }
+
+    <<\Slack\GraphQL\Field('isDeprecated', 'Boolean for whether or not the field is deprecated')>>
+    public function isDeprecated(): bool {
+        return $this->inner_type->isDeprecated();
+    }
+
+    <<\Slack\GraphQL\Field('deprecationReason', 'Reason the field was deprecated')>>
+    public function getDeprecationReason(): ?string {
+        return $this->inner_type->getDeprecationReason();
+    }
+
+}
+
+<<\Slack\GraphQL\ObjectType('__EnumValue', 'Enum value introspection')>>
+final class __EnumValue {
+    public function __construct(private IntrospectableEnumValue $inner_type) {}
+
+    <<\Slack\GraphQL\Field('name', 'Name of the enum value')>>
+    public function getName(): string {
+        return $this->inner_type->getName();
+    }
+
+    <<\Slack\GraphQL\Field('description', 'Description of the enum value')>>
+    public function getDescription(): ?string {
+        return $this->inner_type->getDescription();
+    }
+
+    <<\Slack\GraphQL\Field('isDeprecated', 'Boolean for whether or not the enum value is deprecated')>>
+    public function isDeprecated(): bool {
+        return $this->inner_type->isDeprecated();
+    }
+
+    <<\Slack\GraphQL\Field('deprecationReason', 'Reason the enum value was deprecated')>>
+    public function getDeprecationReason(): ?string {
+        return $this->inner_type->getDeprecationReason();
+    }
+}
+
+<<\Slack\GraphQL\ObjectType('__InputValue', 'Input value introspection')>>
+final class __InputValue {
+    public function __construct(private IntrospectableInputValue $inner_type) {}
+
+    <<\Slack\GraphQL\Field('name', 'Name of the field')>>
+    public function getName(): string {
+        return $this->inner_type->getName();
+    }
+
+    <<\Slack\GraphQL\Field('description', 'Description of the field')>>
+    public function getDescription(): ?string {
+        return $this->inner_type->getDescription();
+    }
+
+    <<\Slack\GraphQL\Field('type', 'Type of the field')>>
+    public function getType(): __Type {
+        return new __Type($this->inner_type->getType());
+    }
+
+    // TODO: Not sure how this is possible for input objects? Maybe if they're a class?
```

Comment:
Could maybe be done via an attribute.

Reply:
Yep something like this works:

```Hack
<<
  GraphQL\InputObjectType('Foo')
  GraphQL\DefaultValues(dict[
    'some_field' => 42,
    'other_field' => '...'
  ])
```

but also we don't have to support default values for now.